BitwardenShared/Core/Platform/Services/API/Requests/EventRequest.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+import Networking
+
+// MARK: - EventRequestModel
+
+/// Data model for the body of an event upload.
+///
+struct EventRequestModel: Codable, Equatable {
+    /// The type of event.
+    let type: EventType
+
+    /// The ID of the cipher related to the event, if there is one.
+    let cipherId: String?
+
+    /// The time when the event occurred.
+    let date: Date
+}
+
+/// Data model for the body of an event upload.
+///
+struct EventRequestBody: JSONRequestBody, Codable, Equatable {
+    let events: [EventRequestModel]
+}
+
+// MARK: - EventRequest
+
+/// A request for uploading organization events.
+struct EventRequest: Request {
+    typealias Response = EmptyResponse
+
+    // MARK: Properties
+
+    /// The body of the request.
+    var body: [EventRequestModel]? { requestBody.events }
+
+    /// The HTTP method of the request.
+    let method: HTTPMethod = .post
+
+    /// The URL path of this request.
+    let path = "/collect"
+
+    /// The actual body of the request.
+    let requestBody: EventRequestBody
```

Comment:
🤔 I'm curious if the `EventRequestBody` is useful here, if we're sending the events as an array (as opposed to wrapped by the body struct). Could this just store the events as an array?

Reply:
I admittedly went through several different variations of this while trying to get "sending an array to the server" to work, but I think there might have been some reason I couldn't? I'll double-check.